File: day12.go
package main

import "fmt"

// Struct is a blueprint that describes the data
type Person struct {
	Name string
	Age  int
}

// Method with value receives (u User).
// Method attaches the function to Struct
// Without (u *User) method operates on the copy of the data, good for reading
func (p Person) Greet() {
	fmt.Printf("Hi, my name is %s and I am %d.\n", p.Name, p.Age)
}

// Method with pointer receiver
// (p *Person) is pointer receiver, operates on a pointer original data.
func (p *Person) SetAge(newAge int) {
	p.Age = newAge
}

func main() {
	person1 := Person{Name: "Alice", Age: 30}
	person1.Greet()

	person1.SetAge(31)

	person1.Greet()
}

File: python/day12.py
# Like traditional oop define a class
class Person:
    # This constructor method run wen a new instance of the class is created.
    # here 'self' is equivalent of the Go 'receiver' (p or *p)
    def __init__(self, name, age):
        self.name = name
        self.age = age

    # Getter method just to read data
    def greet(self):
        print(f"Hi, my name is {self.name} and I am {self.age}")

    def set_age(self, new_age):
        self.age = new_age
# -------------Main code execution ----------------
person1 = Person("Alice", 30)
person1.greet()
person1.set_age(31)
person1.greet()